### PR TITLE
BUG: Categorical of booleans has object .categories

### LINKED ISSUE #46313: BUG: Categorical of booleans has object .categories
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [1]: import pandas as pd
   ...: 
   ...: obs = pd.DataFrame({"int_cat":[1,2,3], "bool_cat":[True,False,False]}, dtype="category")
   ...: obs["bool_cat"].cat.categories.dtype
Out[1]: dtype('O')

In [2]: bools = pd.Series([True, False, True])
   ...: bools.dtype
Out[2]: dtype('bool')

In [3]: bools.astype("category").cat.categories.dtype
Out[3]: dtype('O')

In [4]: pd.__version__
Out[4]: '1.4.1'
```

### Issue Description

I would assume a categorical generated from a boolean array would have `bool` category values, not `object`.

This is weird, but it's admittedly a funny use case. Upstream of https://github.com/theislab/anndata/issues/724

### Expected Behavior

I would have expected the dtype of the categories to be boolean.

### Installed Versions

<details>
```
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.9.10.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Wed Nov 10 22:23:07 PST 2021; root:xnu-7195.141.14~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.4.1
numpy            : 1.21.5
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 22.0.3
setuptools       : 60.5.0
Cython           : 0.29.25
pytest           : 7.0.1
hypothesis       : None
sphinx           : 4.1.2
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 8.1.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : 2022.02.0
gcsfs            : None
matplotlib       : 3.5.1
numba            : 0.55.1
numexpr          : 2.8.1
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 7.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
sqlalchemy       : 1.4.31
tables           : 3.7.0
tabulate         : 0.8.9
xarray           : 0.21.1
xlrd             : 1.2.0
xlwt             : None
zstandard        : None
```

</details>

### PR DESCRIPTION
**Pull request by rxd330**
- [x] closes #46313 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This patch adds a test for checking regression behavior as described in #46313. The bug was reported to be fixed and I could not reproduce the bug in the latest version as Mar.18, 2022. So, only a test was added.

This is my first contribution to the open-source project. Please let me know if something needs to be fixed. Thanks!

### CODE REVIEW

File: pandas/tests/dtypes/test_dtypes.py [hunk 1]
```diff
@@ -1107,3 +1107,15 @@ def test_compare_complex_dtypes():
...
+
+
+def test_categorical_of_booleans_is_boolean():
+    # https://github.com/pandas-dev/pandas/issues/46313
+    df = pd.DataFrame(
+        {"int_cat": [1, 2, 3], "bool_cat": [True, False, False]}, dtype="category"
+    )
+    value = df["bool_cat"].cat.categories.dtype
+    expected = np.dtype(np.bool_)
+    not_expected = np.dtype(np.object_)
```
Comment (jreback) [inline pandas/tests/dtypes/test_dtypes.py:1119]: you don't need the not expected case

File: pandas/tests/dtypes/test_dtypes.py [hunk 2]
```diff
@@ -1107,3 +1107,15 @@ def test_compare_complex_dtypes():
 
     with pytest.raises(TypeError, match=msg):
         df.lt(df.astype(object))
+
+
+def test_categorical_of_booleans_is_boolean():
+    # https://github.com/pandas-dev/pandas/issues/46313
+    df = pd.DataFrame(
+        {"int_cat": [1, 2, 3], "bool_cat": [True, False, False]}, dtype="category"
```
Comment (jreback) [inline pandas/tests/dtypes/test_dtypes.py:1115]: move to pandas/tests/series/accessors/test_cat_accessor.py

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by rxd330** [comment]
Hi there, I moved the code to the specified location and removed the unnecessary line. 

Thanks for giving feedback!

**Review by jreback — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rxd330

**Pull request merged**